library/alloc/src/string.rs

```diff
@@ -410,7 +410,16 @@ pub struct FromUtf8Error {
 /// ```
 #[stable(feature = "rust1", since = "1.0.0")]
 #[derive(Debug)]
-pub struct FromUtf16Error(());
+pub struct FromUtf16Error {
+    kind: FromUtf16ErrorKind,
+}
+
+#[cfg_attr(no_global_oom_handling, allow(dead_code))]
```

Comment:
It's always a good idea to `#[expect]` these sort of things so that people know when they become unnecessary.
```suggestion
#[cfg_attr(no_global_oom_handling, expect(dead_code))]
```